applications/luci-app-p910nd/htdocs/luci-static/resources/view/p910nd.js

```diff
@@ -0,0 +1,215 @@
+// #
+// # Copyright (c) 2024.
+// # Author: systemcrash
+// #
+
+'use strict';
+'require fs';
+'require form';
+'require uci';
+'require ui';
+'require view';
+'require network';
+
+var pkg = {
+	get Name() { return 'p910nd'; },
+	get Description() { return _('Port 910n daemon'); },
+	get URL() { return 'https://openwrt.org/packages/pkgdata/' + pkg.Name + '/'; }
+};
+
+return view.extend({
+
+	option_install_kmod_lp: function() {	
+		window.open(L.url('admin/system/opkg') +
+			'?query=kmod-lp', '_blank', 'noopener');
+	},
+	option_install_kmod_usb: function() {	
+		window.open(L.url('admin/system/opkg') +
+			'?query=kmod-usb-printer', '_blank', 'noopener');
+	},
+
+	load: function () {
+
+		return Promise.all([
+			uci.load(pkg.Name), // data[0]
+			network.getDevices(), // data[1]
+
+			//	# find -L /dev -maxdepth 3 -type c -name lp[0-9]
+			fs.exec_direct('/usr/bin/find', [ '-L', '/dev', '-maxdepth', '3', '-type', 'c', '-name', 'lp[0-9]' ]).catch(function(err) {
+				ui.addNotification(null, E('p', {}, _('Error executing "find" command: ' + err.message)));
+					return '';
+			}).then(data => {return data;}), // data[2]
+
+			fs.exec_direct('/usr/libexec/opkg-call', [ 'list-installed' ], 'text').catch(function(err) {
+				ui.addNotification(null, E('p', {}, _('Error calling "opkg list-installed": ' + err.message)));
+				console.log(err);
+					return '';
+			}).then(data => {return data.trim().split('\n').sort().filter((string) => string.includes(': kmod-'));}), // data[3]
+
+		]);
+	},
+
+	render: function (data) {
+
+		var m, d, s, o;
+
+		const have_kmod_lp = data[3] ? data[3].find((string) => string.includes('kmod-lp')) ? true: false : false;
+		const have_kmod_usb_printer = data[3] ? data[3].find((string) => string.includes('kmod-usb-printer')) ? true: false : false;
+		
+		m = new form.Map(pkg.Name, pkg.Description, _('map ports 9100-9109 to local printers.'));
+		m.readonly = !L.hasViewPermission();
```

Comment:
is this required?

Reply:
Possibly not required, but does not hurt to have it